packages/astro/src/default/components/MobileContentToggle.astro

```diff
@@ -0,0 +1,44 @@
+<div class="h-12 sm:hidden"></div>
+<view-toggle
+  class="fixed sm:hidden h-12 bottom-0 w-full bg-tk-elements-app-backgroundColor border-t border-tk-elements-app-borderColor flex justify-center items-center z-60 text-tk-elements-app-textColor"
+>
+  <button>Tutorial</button>
+  <button class="rounded-full w-8 h-4 p-0.5 bg-gray-4 dark:bg-gray-6 mx-2 relative">
+    <span class="inline-block transition-all absolute top-0.5 left-0.5 rounded-full bg-white dark:bg-gray-2 w-3 h-3"
+    ></span>
+  </button>
+  <button>Editor</button>
+</view-toggle>
+<script>
+  import { viewStore, type View } from '../stores/view-store';
+
+  class ViewToggle extends HTMLElement {
+    constructor() {
+      super();
+
+      const [tutorialBtn, toggleButton, editorButton] = this.querySelectorAll(':scope > button' as 'button');
```

Comment:
Why is the `:scope > button` casted to the string `button`? Does that have any benefit?

Reply:
It does! :smiley:  That was a suggestion from Ari: https://github.com/stackblitz/tutorialkit/pull/91#discussion_r1652969703

Reply:
Yea that's the power of string literal types 🙌 🔥